### PR TITLE
IpMap Test update

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
More tests from Larry related to #2250, #2253.
Conversion to Catch unit test framework.

Other changes to the unit test framework

* `main.cpp` was moved to `main.cc`.
* The file comments for `main.cc` were changed to include better documentation and to use C style comments instead of python style.

### CODE REVIEW

**Comment by SolidWallOfCode** [comment]
[approve ci icc]

**Comment by SolidWallOfCode** [comment]
[approve ci freebsd]

**Comment by SolidWallOfCode** [comment]
[approve ci freebsd]

**Comment by SolidWallOfCode** [comment]
Although I think this can be merged, there is further related work to be done.

* Including `catch.hpp` is annoying because it's in a directory that is not on the standard include paths. Either the test files must do relative paths to reach up and get the file or the makefile must be altered to put it on the paths. I think we can do better.
* This impacts `main.cc` because if it is compiled from local makefile, what include path does it use for `catch.hpp`? Without standard include paths any choice will be wrong somewhere.

File: lib/ts/test/test_IpMap.cc [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#include <ts/IpMap.h>
+#include <../../tests/catch.hpp>
```
Comment (zwoop) [inline lib/ts/test/test_IpMap.cc]: We might want to consider to add tests/ to the search paths (-I) in the various Makefiles.am that will build tests? And then consistently do

```
#include "tests/catch.hpp"
```

**Comment by SolidWallOfCode** [comment]
[approve ci freebsd]

**Comment by SolidWallOfCode** [comment]
Gah. It works on my FreeBSD VM but not this one. I'll need to investigate more.

**Comment by SolidWallOfCode** [comment]
[approve ci freebsd]

**Comment by SolidWallOfCode** [comment]
It's some sort of merge problem. As you can see in the file list, there is a `tests/unit_tests/Makefile.am` file in the PR. However, the `configure.ac` debugging I put in reports during configuration
```
Unit Pre
ls: tests/unit_tests: No such file or directory
ls: tests/unit_tests/.deps: No such file or directory
```
that the file doesn't exist. I'm not sure how to get around this, as even after rebasing with current master the problem persists.

**Pull request merged**